crates/ruff_linter/src/rules/ruff/rules/asyncio_dangling_task.rs

```diff
@@ -105,6 +110,27 @@ pub(crate) fn asyncio_dangling_task(checker: &mut Checker, expr: &Expr) {
     }
 }
 
+pub(crate) fn asyncio_dangling_task_unused(
+    checker: &Checker,
+    scope: &Scope,
+    diagnostics: &mut Vec<Diagnostic>,
+) {
+    for binding in scope
+        .binding_ids()
+        .map(|binding_id| checker.semantic().binding(binding_id))
+        .filter(|binding| binding.kind.is_assignment() && !binding.is_used())
+    {
+        let Some(source) = binding.source else {
+            continue;
+        };
+        let Stmt::Assign(ast::StmtAssign { value, .. }) = checker.semantic().statement(source)
+        else {
+            continue;
+        };
+        asyncio_dangling_task(checker, value, diagnostics);
```

Comment:
Nice job figuring this out.

Reply:
Can you try using `typing::analyze::resolve_assignment` here instead? It will also handle `AnnAssign`.

Reply:
Hmm, not sure how to do it without some refactoring.
I currently use assignment value expression which (if I'm not mistaken) is discarded after processing in `resolve_assignment`?

For now I've extended the pattern match to include AnnAssign as done in [unnecessary_list_cast](https://github.com/asafamr-mm/ruff/blob/00bcfc87c8b5d1380f4902d4019205b2840caac6/crates/ruff_linter/src/rules/perflint/rules/unnecessary_list_cast.rs#L120 ) - but would gladly change it